### DISCUSSION TITLE
nb_attr.h:294:14: warning: ISO C++ forbids zero-size array [-Wpedantic]

### DISCUSSION THREAD

**Discussion by kennethassogba** [wjakob/nanobind#792] [General]
Hello,

I am trying to compile my c++ source files with `-pedantic` gcc compiler flags, but not external libs like nanobind.
I do, however, get this warning:
```bash
nanobind/nb_attr.h: In instantiation of ‘struct nanobind::detail::func_data_prelim<0>’:
nanobind/nb_func.h:191:38:   required from ‘PyObject* nanobind::detail::func_create ...
nanobind/nb_func.h:352:37:   required from ‘void nanobind::cpp_function_def ...
nanobind/nb_class.h:402:33:   required from here
nanobind/nb_attr.h:294:14: warning: ISO C++ forbids zero-size array [-Wpedantic]
  294 |     arg_data args[Size];
```
My compilation options are set to Private. However, since nb_attr.h is included in my sources, the pedantic flag applies to it.
```cmake
add_library(demeter ${DEM_SOURCES})
target_compile_options(demeter PRIVATE -Wall -Wextra -pedantic)
...
nanobind_add_module(demeter_ext
                    NB_DOMAIN demeter
                    src/demeter/model/material.bind.cpp)
target_link_libraries(demeter_ext PRIVATE demeter)
```
Ideally, I would not have to recompile nanobind, but I use it as a submodule.
Do you plan to use the same trick as for MSVC a few lines above in nb_attr.h?
```cpp
#if defined(_MSC_VER)
    // MSVC doesn't support zero-length arrays
    arg_data args[Size == 0 ? 1 : Size];
#else
    // GCC and Clang do.
    arg_data args[Size]; // <- [-Wpedantic]
#endif
```

**Comment by wjakob** [comment]
Did you read the comment some lines above?

**Reply by kennethassogba** [reply]
Just read it, thank you very much.

**Reply by wojdyr** [reply]
the comment is easy to overlook if it's not on the same line or directly before
https://github.com/wjakob/nanobind/issues/639#issuecomment-2239059010
